components/modals/sign-in-modal.tsx

```diff
@@ -9,12 +11,28 @@ import GithubColor from '@/components/icons/github-color';
 import { Dialog, DialogCloseButton, DialogContent, DialogHeader } from '@/components/ui/dialog';
 
 export const SignInModal = () => {
+  // States for holding is loading operations, in this scenario the
+  // sign in state (is signing in: true => signed in: false)
+  const [isPending, startTransition] = useTransition();
+
+  // Extracts params from useModal hook
   const { isOpen, onClose, type } = useModal();
 
   const isModalOpen = isOpen && type === 'signIn';
 
+  const onSignIn = () => {
+    // Start the sign in process
+    startTransition(() => {
+      // Sign in using github oath app and redirect to the page that
+      // the user is currently on
+      signIn('github', {
+        callbackUrl: window.location.pathname,
```

Comment:
I suggest using `useRouter` instead of `window.location.pathname` to get the current path here.

> It provides a more reliable and consistent way to access the current path compared to directly using `window.location.pathname`.

Reply:
Done! But I use `usePathname` hook instead of `useRouter`, both works the same.

![image](https://github.com/projectopia/website/assets/118145239/81246dc9-88ae-48e2-8a85-f2a10f37b9a2)

Reply:
I think @thangbuiq will handle this task. I will put the required env variables in `.env.example` for him to handle.